### PR TITLE
Prevent user from sending negative sol

### PR DESCRIPTION
**Pull request by ana2201**
## Description
Prevent the user from sending negative SOL

### Preview
https://forum-staging-flliae9yb-usedispatch.vercel.app/

https://www.loom.com/share/e6e5af154f75422f870e6207cb806967

### CODE REVIEW

File: src/components/forums/topic/GiveAward.tsx [hunk 1]
```diff
@@ -118,17 +161,20 @@ export function GiveAward(props: GiveAwardProps) {
                 name="award"
```
Comment (zfaizal2) [inline src/components/forums/topic/GiveAward.tsx:161]: can you replace this component with the Input.tsx component as well in `components/common`

**Review by zfaizal2 — CHANGES_REQUESTED** [review]

**Comment by viksit** [comment]
image


I can go in to the form, put cursor in
press left arrow
add -1

solution: make sure the button to send does not activate if value in form is negative.

File: .eslintrc.json [hunk 2]
```diff
@@ -31,21 +30,30 @@
...
             {
-                "checksVoidReturn":  {
+                "checksVoidReturn": {
                     "attributes": false
-                  }
+                }
+            }
+        ],
+        "@typescript-eslint/no-shadow": "off",
+        "@typescript-eslint/strict-boolean-expressions": [
```
Comment (zfaizal2) [inline .eslintrc.json:51]: we should leave this rule on and continue to do null checks manually.

File: src/components/common/Input.tsx [hunk 3]
```diff
@@ -1,66 +1,60 @@
...
+    setTextValue(Number.isNaN(value) ? '' : value);
+  }, [value]);
 
   return (
     <>
       <input
         type="text"
         placeholder={placeholder}
-        step={type === 'number' ? step || 'any' : undefined}
+        step={type === 'number' ? step ?? 'any' : undefined}
```
Comment (zfaizal2) [inline src/components/common/Input.tsx:57]: what is this meant to do?

**Review by zfaizal2 — APPROVED** [review]

**Pull request merged**